askama_parser/src/lib.rs

```diff
@@ -252,8 +251,88 @@ fn bool_lit(i: &str) -> ParseResult<'_> {
 fn num_lit(i: &str) -> ParseResult<'_> {
     recognize(tuple((
         opt(char('-')),
-        digit1,
-        opt(pair(char('.'), digit1)),
+        // digit1,
```

Comment:
Looks like there's a bunch of duplication here, which I'd like to avoid. I'm guessing this will require some named parsers, which would probably aid in making this code easier to understand. Also please remove the commented out version of the old code.